### PR TITLE
Get rid of dep in favour of go mod

### PR DESCRIPTION
**Pull request by rojkov**
This is going to increase build time in CI significantly due to deps required to be downloaded for every build.

Also bump up K8s dependency to v1.15.3.

Closes: #202

### CODE REVIEW

File: build/docker/intel-fpga-admissionwebhook.Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-FROM golang:1.11 as builder
-ARG DIR=/go/src/github.com/intel/intel-device-plugins-for-kubernetes
+FROM golang:1.12 as builder
```
Comment (mythi) [inline build/docker/intel-fpga-admissionwebhook.Dockerfile]: This changes in #199 so both fpga Dockerfile changes can be dropped.

File: build/docker/intel-qat-plugin.Dockerfile [hunk 2]
```diff
@@ -34,10 +34,10 @@ RUN test -z "${TAGS_KERNELDRV}" \
     && cd /usr/src/qat/quickassist/utilities/adf_ctl \
     && make KERNEL_SOURCE_DIR=/usr/src/qat/quickassist/qat \
     && install -D adf_ctl /install_root/usr/local/bin/adf_ctl )
-ARG DIR=/go/src/github.com/intel/intel-device-plugins-for-kubernetes
+ARG DIR=/intel-device-plugins-for-kubernetes
 WORKDIR $DIR
 COPY . .
-RUN cd cmd/qat_plugin; go install ${TAGS_KERNELDRV}
+RUN cd cmd/qat_plugin; echo "build tags: ${TAGS_KERNELDRV}"; go install -tags "${TAGS_KERNELDRV}"
```
Comment (mythi) [inline build/docker/intel-qat-plugin.Dockerfile]: agree this is better. my original idea was `TAGS_KERNELDRV="-tags kerneldrv"`

File: build/docker/intel-fpga-admissionwebhook.Dockerfile (hunk 1 above)
Comment (rojkov) [reply build/docker/intel-fpga-admissionwebhook.Dockerfile]: #199 doesn't update `ARG DIR=/go/src/github.com/intel/intel-device-plugins-for-kubernetes` which implicitly switches go mod off.

**Comment by rojkov** [comment]
> Perhaps add "Closes: #202"

Thanks! Added.

> How much is 'significantly'? `GOPROXY` might be an option for Jenkins at least?

Jenkins is not a problem. TravisCI is. It has a hard time limit of 40 mins and we often hit this limit already. I was trying to say that we might have to disable some tests, e.g. `make demos` or `buildah` builds.

File: build/docker/intel-fpga-admissionwebhook.Dockerfile (hunk 1 above)
Comment (kad) [reply build/docker/intel-fpga-admissionwebhook.Dockerfile]: please update this PR based on the merged changes from PR #199

Comment (rojkov) [reply build/docker/intel-fpga-admissionwebhook.Dockerfile]: Ok. Resolved the conflict and force-pushed.

**Comment by mythi** [comment]
Would this work: https://docs.travis-ci.com/user/caching/

**Comment by rojkov** [comment]
@mythi This looks like the right thing indeed. We might want to cache `$GOPATH/pkg/mod`.

@kad could you please check if "Build pushed branches" is on in the repo's settings.

**Comment by kad** [comment]
@rojkov  it is "on". you can see it from here: https://travis-ci.org/intel/intel-device-plugins-for-kubernetes/branches

File: .travis.yml [hunk 3]
```diff
@@ -5,11 +5,16 @@ dist: xenial
...
+cache:
+  directories:
+    - $GOPATH/pkg/mod
+
 env:
   global:
     - RUNC_VERSION=v1.0.0-rc8
     - CRIO_VERSION=v1.14.6
     - BUILDAH_VERSION=v1.10.0
+    - GO111MODULE=on
```
Comment (kad) [inline .travis.yml]: maybe let's cleanup also "## hack for building on forks" ?

Comment (bart0sh) [reply .travis.yml]: It's very handy and I'm using it regularly for building my own fork.

**Comment by kad** [comment]
@rojkov please squash some of the commits. e.g. group all CI (travis and jenkins) into one commit.

File: .travis.yml (hunk 3 above)
Comment (rojkov) [reply .travis.yml]: What is it for? The travis build seems to proceed successfully without this hack?

**Comment by rojkov** [comment]
Ok, squashed.

File: .travis.yml (hunk 3 above)
Comment (bart0sh) [reply .travis.yml]: It's for building in a forked project. Without it it was not possible to build as far as I remember.

**Comment by mythi** [comment]
The execution time did not improve too much. The container image builds do not get to see the cache at least

**Comment by rojkov** [comment]
I understood it so that the cache is stored to S3 only for branch builds, not PRs. It's going to benefit new PRs after this one is merged.

And image builds won't benefit from this. I'm experimenting with making the cache available in build containers.

**Pull request merged**